### PR TITLE
fix: PodMonitor regression

### PR DESCRIPTION
**Pull request by CiraciNicolo**
# Summary

This pr fix a regression introduced by the new Plugin API which caused the PodMonitor - and probably other CRD - to not be created. Some E2E test have been introduced to detect issues with PodMonitors.

Furthermore some commodities have been added and general code tweaks.

# Additional Info

Kindly note that this pr is a cherry picked version of https://github.com/cloudnative-pg/cloudnative-pg/pull/4268, which contains a broader review of PodMonitor specs.

Fixes: #4286

### CODE REVIEW

File: hack/setup-cluster.sh [hunk 1]
```diff
@@ -614,6 +620,12 @@ deploy() {
   echo "${bright}Done deploying manifests from current worktree on cluster ${CLUSTER_NAME}${reset}"
 }
 
+reload() {
```
Comment (phisco) [inline hack/setup-cluster.sh:623]: This is not strictly related to this fix, please let's keep the pr as simple as possible.

File: hack/watch-cluster.sh
Comment (phisco) [inline hack/watch-cluster.sh:1]: This is not strictly related to this fix, nor directly used anywhere, please let's keep the pr as simple as possible.

File: hack/setup-cluster.sh (hunk 1 above)
Comment (CiraciNicolo) [reply hack/setup-cluster.sh:623]: Both this function and the watch-cluster.sh have been used to implements the e2e test, they are pretty related to the development; should I open a PR for this two file? They would be used for other contributions.

Comment (phisco) [reply hack/setup-cluster.sh:623]: It's usually better to keep PRs focused at first, to both help the maintainers reviewing and the author increasing the chances to see their contribution merged quickly. I'd suggest you to keep them for a future dedicated contribution.

Comment (CiraciNicolo) [reply hack/setup-cluster.sh]: Removed then :)

**Comment by phisco** [comment]
Can you open an issue describing the bug you are fixing a bit more in details?

**Comment by rouke-broersma** [comment]
@phisco this is a blocker for podMonitor, having podMonitor enabled on the Cluster CRD at all causes reconciliation to fail.

**Comment by Ornias1993** [comment]
> Can you open an issue describing the bug you are fixing a bit more in details?

@phisco The fact you just released a new Helm Chart version with the bug included, knowing it was going to break things for all users using podmonitors, is completely unacceptable.

**Comment by leonardoce** [comment]
@CiraciNicolo, thank you for looking at this.

About https://github.com/cloudnative-pg/cloudnative-pg/pull/4282/commits/301836deed184c80cef0ff65ce40a02cc334249b#diff-bc37d034bad564583790a46f19d807abfe519c5671395fd494d8cce506c42947R40-R43 I think you can add these lines into `.git/info/exclude` (I'm an asdf user too and I do the same).

And I agree with https://github.com/cloudnative-pg/cloudnative-pg/pull/4282/commits/d527a325568de048f52d6c48df36707142250a17#diff-1ca08b6f9e9610ef14189184290a644a29adbb47e6f8653a479d75fdce4a8452L125 too. Can you file a PR about that?

I removed it from this PR to ease the review process and get it merged earlier.

**Comment by leonardoce** [comment]
I ran an E2e test suite for it this morning in the EDB repo (to save some OSS workers), and they're fine.

 https://github.com/EnterpriseDB/cloudnative-pg/actions/runs/8873803689

**Comment by CiraciNicolo** [comment]
Hi @leonardoce,
thanks for looking into this PR.

> And I agree with https://github.com/cloudnative-pg/cloudnative-pg/commit/d527a325568de048f52d6c48df36707142250a17#diff-1ca08b6f9e9610ef14189184290a644a29adbb47e6f8653a479d75fdce4a8452L125 too. Can you file a PR about that?

Are you talking about the defer refactoring? If so, I will open a PR ASAP 😊

**Comment by leonardoce** [comment]
> Are you talking about the defer refactoring? If so, I will open a PR ASA
Great. And assign it to me

**Comment by CiraciNicolo** [comment]
PR opened, but I'm afraid I cannot assign issues.

**Review by sxd — APPROVED** [review]

**Review by leonardoce — APPROVED** [review]

**Comment by leonardoce** [comment]
I run a full E2e test suite with cloud environments too: https://github.com/EnterpriseDB/cloudnative-pg/actions/runs/8878558753
